tests/test_brain_large_speedup.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""Test improved calculation of log_zinb_positive and log_nb_positive for better performance on brain_large."""
+
+from scvi.dataset import SyntheticDataset
+import torch
+import torch.nn.functional as F
+
+batch_size = 128
+nb_genes = 720
+n_batches = 1
+
+
+synthetic_dataset = SyntheticDataset(batch_size=batch_size, nb_genes=nb_genes,
+                                     n_batches=n_batches)
+x = torch.from_numpy(synthetic_dataset.X)
+mu = torch.rand(batch_size, nb_genes)
+theta = torch.rand(batch_size, nb_genes)
+pi = torch.rand(batch_size, nb_genes)
+eps = 1e-2
+
+
+def test_log_zinb_positive():
+    """Test that the new method to compute log_zinb_positive is the same as the existing."""
+
+    def existing_method(x, mu, theta, pi, eps=1e-8):
+        case_zero = (F.softplus((- pi + theta * torch.log(theta + eps) - theta * torch.log(theta + mu + eps))) -
+                     F.softplus(-pi))
+
+        case_non_zero = - pi - F.softplus(-pi) + \
+            theta * torch.log(theta + eps) - \
+            theta * torch.log(theta + mu + eps) + \
+            x * torch.log(mu + eps) - \
+            x * torch.log(theta + mu + eps) + \
+            torch.lgamma(x + theta) - \
+            torch.lgamma(theta) - \
+            torch.lgamma(x + 1)
+
+        res = torch.mul((x < eps).type(torch.float32), case_zero) + \
+            torch.mul((x > eps).type(torch.float32), case_non_zero)
+
+        return torch.sum(res, dim=-1)
+
+    def new_method(x, mu, theta, pi, eps=1e-8):
```

Comment:
The unit test is suppose to test the actual code (the code in under the `scvi` directory). Here it looks like the test is just testing the test. @triyangle @romain-lopez  Am I misunderstanding what's going on here?

Reply:
So to write a unit test, you don't general duplicate code. You call the code you want to test that is in the `scvi` directory directly, rather than copying that code to the test file. Does that make sense?

Reply:
Based on my understanding, I believe you're saying we should be calling the functions I've modified directly from `scvi` instead of copying the new implementation over? That makes sense to me, but would the test still have the old implementation to compare the results between the new and old implementations then? For reference I based this test on [test_remove_empty](https://github.com/YosefLab/scVI/blob/master/tests/test_remove_empty.py).